superpeer/handler/MenuHandler.py

```diff
@@ -22,11 +29,77 @@ def serve(self, choice: str) -> None:
 		elif choice == "QUER":
 			pass
 
+		elif choice == "LISTSUPERPEERS":
+
+			shell.print_green('\nList of known peers:')
+
+			for count, friend in enumerate(LocalData.get_super_friends(), start=1):
+				friend_ip4 = LocalData.get_super_friend_ip4(friend)
+				friend_ip6 = LocalData.get_super_friend_ip6(friend)
+				friend_port = LocalData.get_super_friend_port(friend)
+				shell.print_blue(
+					f'{count}] {friend_ip4} {friend_ip6} {str(friend_port)}')
+
 		elif choice == "LISTPEERS":
-			pass
+
+			try:
+				conn = database.get_connection(db_file)
+				conn.row_factory = database.sqlite3.Row
+
+			except database.Error as e:
+				print(f'Error: {e}')
+				return "The server has encountered an error while trying to serve the request."
+
+			try:
+				peer_list = peer_repository.find_all(conn)
+
+				if peer_list is None:
```

Comment:
Dovrebbe esserci la funzione ```empty()``` built-in di Python per controllare se una lista è vuota. Se la ```find_all``` restituisce una lista vuota, questa è diversa da ```None```

Reply:
la soluzione è scrivere:
if not peer_list: 
          ....